### ISSUE TITLE
Foreign key for "X" is missing on "Y"

### ISSUE THREAD

**Issue by eacdev** [RobinBuschmann/sequelize-typescript#757]
**Versions**

- sequelize: 5.21.4
- sequelize-typescript: 1.1.0
- typescript: 3.8.3

**I'm submitting a ...** 
[ x] bug report

**Actual behavior:**
```(node:29538) UnhandledPromiseRejectionWarning: Error: Foreign key for "Server" is missing on "User".```

**Expected behavior:**
The foreign key should be detected.

**Steps to reproduce:**
```ts
export default class Server extends BaseModel<Server> {
  @HasMany(() => User)
  users: User[];
}
```

```ts
export default class User extends BaseModel<User> {
  ...
  @BelongsToMany(
    () => Server,
    () => ServerUser
  )
  servers: Server[];
  ... 
  }
```

```ts
export default class ServerUser extends BaseModel<ServerUser> {
  @ForeignKey(() => User)
  @Column(DataType.BIGINT)
  userId: number;

  @ForeignKey(() => Server)
  @Column(DataType.BIGINT)
  serverId: number;
}
```

**Related code:**
https://github.com/eacdev/express-typescript-vue-app/commit/c5ef23961f0aabc5f7651c95b0cd80a06b6d3eb5

Thanks!

**Comment by eacdev** [comment]
After noticing that the docs. say to put both Server and User as "belongs to many", I did it, and the problem was fixed.

But, isn't it technically that a Server "has many" Users and a User "belongs to many" Servers?
